### PR TITLE
PIC-3787 Re-Add Trivy and veracode scanning of main branch

### PR DESCRIPTION
**Pull request by Obsiye**
This is to re-add this branch https://github.com/ministryofjustice/crime-portal-gateway/pull/177

----

Add trivy and veracode scanning with the help of [hmpps circleci orbs](https://github.com/ministryofjustice/hmpps-circleci-orb/tree/main/src/jobs)

Schedule a trivy and veracode security **scanning of our main branch every weekday morning at 7am**

**Other needed changes:**
- Upgrade to Gradle 8.5
- Upgrade to JVM 21
- Add Kotlin-JVM plugin
- Upgrade AWS SDK versions
- Upgrade kotlin("plugin.spring")
- Fix docker-compose localstack by updating the volumes location
    -   More info here: https://docs.localstack.cloud/references/filesystem/#localstack-volume

- Update the KTLint version
- Fix KTLint linting issues
- Update AWS SDK version to fix broken test

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -127,3 +127,12 @@ workflows:
           executor:
             name: hmpps/java
             tag: "19.0"
```
Comment (sameermoj) [inline .circleci/config.yml:129]: Should this be 21 as well ?

File: docker-compose.yml [hunk 2]
```diff
@@ -21,14 +21,14 @@ services:
     container_name: localstack
     ports:
       - "4566-4597:4566-4597"
-      - 8999:8080
-      - 9080:9080
+      - "8999:8080"
+      - "9080:9080"
```
Comment (sameermoj) [inline docker-compose.yml:25]: The only port that needs exposing is 4566

File: build.gradle.kts [hunk 3]
```diff
@@ -1,17 +1,26 @@
-import org.gradle.internal.impldep.org.junit.experimental.categories.Categories.CategoryFilter.exclude
-
 plugins {
     id("uk.gov.justice.hmpps.gradle-spring-boot") version "4.9.0"
-    kotlin("plugin.spring") version "1.8.0"
+    kotlin("plugin.spring") version "1.9.0"
```
Comment (markreesmoj) [inline build.gradle.kts:3]: Might be an idea to go straight to version 2 and upgrade the `uk.gov.justice.hmpps.gradle-spring-boot` then you won't need to add lines 8-15

File: build.gradle.kts [hunk 4]
```diff
@@ -1,17 +1,26 @@
-import org.gradle.internal.impldep.org.junit.experimental.categories.Categories.CategoryFilter.exclude
-
```
Comment (markreesmoj) [inline build.gradle.kts:2]: ❤️

File: build.gradle.kts (hunk 3 above)
Comment (markreesmoj) [reply build.gradle.kts:3]: https://github.com/ministryofjustice/hmpps-person-record/blob/cc31fd159bf29480a795edac149ea9c4197d58cf/build.gradle.kts#L12

File: .circleci/config.yml (hunk 1 above)
Comment (markreesmoj) [reply .circleci/config.yml:129]: maybe even replace with this:

```suggestion
        - hmpps/gradle_owasp_dependency_check:
          slack_channel: probation_in_court_alerts_prod
          context:
            - hmpps-common-vars
          jdk_tag: "21.0"
          cache_key: "v3_0"
```

And you can delete the reference to `owasp: entur/owasp@0.0.10` orb

But then, you would have to start looking at the OWASP failures :-)
🐝

File: .circleci/config.yml [hunk 5]
```diff
@@ -129,7 +129,17 @@ workflows:
               only:
                 - main
     jobs:
-      - owasp/gradle_owasp_dependency_check:
-          executor:
-            name: hmpps/java
-            tag: "19.0"
```
Comment (markreesmoj) [inline .circleci/config.yml:135]: You can delete `owasp: entur/owasp@0.0.10` off line 4 now

File: docker-compose.yml [hunk 6]
```diff
@@ -21,14 +21,12 @@ services:
     container_name: localstack
     ports:
       - "4566-4597:4566-4597"
```
Comment (markreesmoj) [inline docker-compose.yml:23]: just 4566 needs to be exposed

**Review by markreesmoj — CHANGES_REQUESTED** [review]
https://github.com/ministryofjustice/dps-gradle-spring-boot/blob/main/release-notes/6.x.md#upgrade-to-kotlin-200

This strongly indicates that upgrading the gradle-spring-boot plugin should be done when going to Kotlin v2 as this PR is doing

**Pull request closed without merging**